backend/src/index.js

```diff
@@ -1,6 +1,6 @@
 const express = require("express");
 const app = express();
-const PORT = 8080;
+const PORT = process.env.PORT || 8080;
```

Comment:
this was the fix for heroku. it seems that heroku passes us the port we should be listening to in environment variables.